media-gfx/graphviz/graphviz-2.40.1.recipe

```diff
@@ -0,0 +1,207 @@
+shapes."
+HOMEPAGE="https://www.graphviz.org/"
+COPYRIGHT="1994-2017 AT&T Corp."
+LICENSE="EPL v1.0"
+REVISION="1"
+SOURCE_URI="https://www.graphviz.org/pub/graphviz/stable/SOURCES/graphviz.tar.gz"
+CHECKSUM_SHA256="ca5218fade0204d59947126c38439f432853543b0818d9d728c589dfe7f3a421"
+SOURCE_FILENAME="graphviz-$portVersion.tar.gz"
+PATCHES="graphviz-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+graphvizLibs="\
+	libcdt \
+	libcgraph \
+	libgvc \
+	libgvpr \
+	liblab_gamut \
+	libpathplan \
+	libxdot \
+	"
+libcdtVersion="5.0.0"
+libcgraphVersion="6.0.0"
+libgvcVersion="6.0.0"
+libgvprVersion="2.0.0"
+liblab_gamutVersion="1.0.0"
+libpathplanVersion="4.0.0"
+libxdotVersion="4.0.0"
+portVersionCompat="$portVersion compat >= 2"
+for i in $graphvizLibs; do
+	eval "${i}VersionCompat=\"\$${i}Version compat >= \${${i}Version%%.*}\""
+done
+
+PROVIDES="
+	graphviz$secondaryArchSuffix = $portVersionCompat
+	cmd:acyclic$secondaryArchSuffix
+	cmd:bcomps$secondaryArchSuffix
+	cmd:ccomps$secondaryArchSuffix
+	cmd:circo$secondaryArchSuffix
+	cmd:cluster$secondaryArchSuffix
+	cmd:dijkstra$secondaryArchSuffix
+	cmd:dot_builtins$secondaryArchSuffix
+	cmd:dot$secondaryArchSuffix
+	cmd:dot2gxl$secondaryArchSuffix
+	cmd:edgepaint$secondaryArchSuffix
+	cmd:fdp$secondaryArchSuffix
+	cmd:gc$secondaryArchSuffix
+	cmd:gml2gv$secondaryArchSuffix
+	cmd:graphml2gv$secondaryArchSuffix
+	cmd:gv2gml$secondaryArchSuffix
+	cmd:gv2gxl$secondaryArchSuffix
+	cmd:gvcolor$secondaryArchSuffix
+	cmd:gvgen$secondaryArchSuffix
+	cmd:gvpack$secondaryArchSuffix
+	cmd:gvpr$secondaryArchSuffix
+	cmd:gvmap$secondaryArchSuffix
+	cmd:gvmap.sh$secondaryArchSuffix
+	cmd:gxl2dot$secondaryArchSuffix
+	cmd:gxl2gv$secondaryArchSuffix
+	cmd:mm2gv$secondaryArchSuffix
+	cmd:neato$secondaryArchSuffix
+	cmd:nop$secondaryArchSuffix
+	cmd:osage$secondaryArchSuffix
+	cmd:patchwork$secondaryArchSuffix
+	cmd:prune$secondaryArchSuffix
+	cmd:sccmap$secondaryArchSuffix
+	cmd:sfdp$secondaryArchSuffix
+	cmd:tred$secondaryArchSuffix
+	cmd:twopi$secondaryArchSuffix
+	cmd:unflatten$secondaryArchSuffix
+	lib:libcdt$secondaryArchSuffix = $libcdtVersionCompat
+	lib:libcgraph$secondaryArchSuffix = $libcgraphVersionCompat
+	lib:libgvc$secondaryArchSuffix = $libgvcVersionCompat
+	lib:libgvpr$secondaryArchSuffix = $libgvprVersionCompat
+	lib:liblab_gamut$secondaryArchSuffix = $liblab_gamutVersionCompat
+	lib:libpathplan$secondaryArchSuffix = $libpathplanVersionCompat
+	lib:libxdot$secondaryArchSuffix = $libxdotVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	graphviz${secondaryArchSuffix}_devel = $portVersion
```

Comment:
> Still builds ok for me (for x86 on x86_gcc2) ...

@Begasus: I haven't investigated yet why it's not building for me on x86, but since it is building fine for you and since the recipe looks OK (and builds fine for me on x86_64), I'd say we can probably merge it.

Just one thing before the merge. @alaviss, should we add a compat for the _devel package?
```diff
 PROVIDES_devel="
-	graphviz${secondaryArchSuffix}_devel = $portVersion
+	graphviz${secondaryArchSuffix}_devel = $portVersionCompat
```
Thanks!

Reply:
I would prefer not having them at all. As our policy requires explict dependencies declaration, soversion becomes the prefered number to compare against. So unless there exists any port uses this information, I would recommend against it.